src/github.com/couchbase/sync_gateway/auth/user.go

```diff
@@ -133,6 +136,23 @@ func (user *userImpl) validate() error {
 	return nil
 }
 
+func (user *userImpl) validatePassword() bool {
+
+	// if it's an anon user, they should not have a password
+	if user.Name_ == "" {
+		return (user.PasswordHash_ == nil)
+	}
+
+	// if empty passwords are allowed, skip validation
+	if user.AllowEmptyPassword() {
+		return true
```

Comment:
So ... it takes a _lot_ of code changes just to plumb that boolean down to this point where you can skip the validation. And you end up persisting the allowEmptyPassword flag in the user account, which makes me uncomfortable since this is just supposed to be a temporary override for testing.

What I'd suggest instead is taking the password validation out of the User class, and moving it up to the db package where it's handing updating a user account. At that point the allowEmptyPassword flag is a lot easier to reach and can be made in-memory only.